src/frontend/utils/index.ts

```diff
@@ -1,6 +1,13 @@
 import React from "react";
+import { Observable } from "rxjs";
 
 // takes a promise, and returns a react hook for it.
+// use like this:
+//
+// export const isAvailable = (db: dbTypes.LoadedDb): Promise<boolean> => Promise.resolve(true);
+//
+// const maybeAddNote = promiseAsHook(note.isAvailable(db)) && "it's available!";
+//
 export const promiseAsHook = <T>(p: Promise<T>) => {
```

Comment:
same neuroticism: `usePromise()`?

Reply:
I like it, `usePromise` and `useObservable`